test/integration/puppetlabs/services/certificate_authority/certificate_authority_int_test.clj

```diff
@@ -349,3 +350,54 @@
+    (logutils/with-test-logging
+      (bootstrap/with-puppetserver-running-with-mock-jrubies
+       "JRuby mocking is safe here because all of the requests are to the CA
+       endpoints, which are implemented in Clojure."
+        app
+        ;; Q: Does one need to make up other config params like for eg. infra-crl-path
+        ;; or can we assume anything unspecified will exist with appropriate defaults
+        ;; Would the below override whatever defaults?
+        {:certificate-authority {:compile-masters ["dummy-compile-master"]}}
+        (testing "should update infrastructure CRL"
+          (let [ca-cert (bootstrap/get-ca-cert-for-running-server)
+                subject "dummy-compile-master"
+                ;; In one of the other unit tests I noticed the localhost cert is
+                ;; being revoked. If that does not affect things we could do that instead of 
+                ;; creating a new one. But wondering it actually may have impact on other 
+                ;; tests since this is integration
+                cm-cert (bootstrap/get-cert-signed-by-ca-for-running-server
+                             ca-cert
+                             subject)
+                 ;; TODO - Extract serial number of abvoe cert. Is there an easy way
+                 ;; to do it here or can we make assumption on the expected serial number
+                 ;;
+
+                 ;; Q: Do we need to do something to explicitly moved the certificate
+                 ;; to CA's signed folder
+                 cert-status-request (fn [action]
+                                       (http-client/put
+                                         (str "https://localhost:8140/"
+                                              "puppet-ca/v1/certificate_status/"
+                                              subject)
+                                         (merge options
+                                                {:body (str "{\"desired_state\": \""
+                                                            action
+                                                            "\"}")
+                                                 :headers {"content-type"
+                                                           "application/json"}}))) ]
+            (testing "Infra CRL should contain the revoked cert"
+               (let [revoke-response (cert-status-request "revoked")]
+                 ;; If the revocation was successful we want to check the infra CRL contains above certificate
+                 ;; Can I just change the below to have an AND check as I have attempted below
+                 ;; Q: How do I access CaSettings or just certificate-authority settings here to look up infra-crl-path?  
+                 (is (and (= 204 (:status revoke-response)) (utils/revoked? (utils/pem->crl (:infra-crl-path settings)) cert)))))
+          )))))
+    ;; TODO/Open - Should we also have an other way round test like if revoke a non compile master
```

Comment:
Yeah, I don't think it would be bad in an integration test to revoke a compiler and non-compiler and then check both crls to ensure the correct content. More granular than that and I'd move to a lower level of testing.